project/src/Sound.java

```diff
@@ -77,15 +85,160 @@ public Sound(String sound_file)throws Exception{
         if(this.sound_file.exists()){
             this.init(sound_file);
         }else{
-            throw new Exception("File not found");
+            throw new Exception("Errors occured while opening the file");
         }
     }
     
-    private void init(String sound_file){
+    private void init(String sound_file)throws Exception{
         this.absolute_path = this.sound_file.getAbsolutePath();
         this.sound_size = (int) (this.sound_file.length()/MB_SIZE);
-        Mp3File work_file = new Mp3File(sound_file);
         
+        //We start using the Mp3 library for this work
+        //We will find all the other informations concerning the file
+            
+        this.work_file = new Mp3File(sound_file);
+        
+        
+        //now we get all the key elements of the MP3 file
+        
+        if(this.work_file.hasId3v2Tag()){
+            
+           this.file_tag = this.work_file.getId3v2Tag();
+           this.album_name = file_tag.getAlbum();
+           this.artist_name = file_tag.getArtist();
+           this.sound_duration = file_tag.getLength();
+           this.sound_name = file_tag.getTitle();
+           this.sound_year = Integer.parseInt(file_tag.getYear());
+           this.sound_genre = file_tag.getGenre();
+           this.sound_publisher = file_tag.getPublisher();
+           this.sound_lyrics = file_tag.getLyrics();
+           this.sound_format = 0;
+           this.user_comments = file_tag.getComment();
+            
+        }  
+        
+    }
+    
+    //setters
+
+    public void setSound_publisher(String sound_publisher) {
+        this.sound_publisher = sound_publisher;
+        save_file();
+    }
+
+    public void setSound_lyrics(String sound_lyrics) {
+        this.sound_lyrics = sound_lyrics;
+        save_file();
+    }
+
+    public void setSound_name(String sound_name) {
+        this.sound_name = sound_name;
+        save_file();
+    }
+
+    public void setArtist_name(String artist_name) {
+        this.artist_name = artist_name;
+        save_file();
+    }
+
+    public void setAlbum_name(String album_name) {
+        this.album_name = album_name;
+        save_file();
+    }
+
+    public void setSound_genre(int sound_genre) {
+        this.sound_genre = sound_genre;
+        save_file();
+    }
+
+    public void setUser_comments(String user_comments) {
+        this.user_comments = user_comments;
+        save_file();
+    }
+    
+    //This function saves all modifications that could have happend on the file
+    //It takes the file_tag and attributes all the setters
+    
+    private void save_file(){
```

Comment:
The save file document will help to directly save the updates on the file tag after each modification of the file. This method may seem time-consuming as each element update causes a file save. To reformat!